spec: add CatalogDatasetFacet

mobuchowski opened the pull request:
Introduces a new CatalogDatasetFacet to the OpenLineage spec. 

More relevant discussion: [issue #3642](https://github.com/OpenLineage/OpenLineage/issues/3642) and [pull request #3639](https://github.com/OpenLineage/OpenLineage/pull/3639).
In short, we don't handle Iceberg tables accessed via JDBC-based or custom catalogs in Spark. Additionally, we don't capture catalog name - and ​this is how users most frequently interact with those tables.

Previously, the approach in PR #3639 attempted to handle JDBC catalogs by creating symlinks with non-compliant dataset information, which was not ideal. The new `CatalogDatasetFacet` standardizes way to include catalog-specific metadata in the lineage events without relying on symlinks.

Drawback of this approach is that consumers who wish to use this information as a dataset identifier need to look at one additional facet: not only top-level dataset name and namespace and symlink facet, but also this one.​

dolfinus commented on client/python/openlineage/client/generated/catalog_dataset.py:
```diff
@@ -0,0 +1,29 @@
+
+
+@attr.define
+class CatalogDatasetFacet(DatasetFacet):
```
Java classes are generated while building .jar, am I right?

mobuchowski replied:
Yes.

We could move this to runtime and add pre-commit checks, to make those changes more visible in PRs.

dolfinus replied:
This is just a question :)

dolfinus commented on spec/facets/CatalogDatasetFacet.json:
```diff
@@ -0,0 +1,51 @@
+              "example": "jdbc, glue, polaris"
+            },
+            "uri": {
+              "description": "URI or connection string to the catalog, if applicable.",
```
This description means that uri can be not applicable to some catalogs, but this field is required in JSON schema

dolfinus commented on spec/facets/CatalogDatasetFacet.json:
```diff
@@ -0,0 +1,51 @@
+              "type": "string",
+              "example": "iceberg, delta, hive"
+            },
+            "source": {
```
Well, OpenLineage events already have processing engine facet, what's the difference?

mobuchowski commented:
@dolfinus @pawel-big-lebowski I've updated the proposed spec:
- renamed `technology` to framework, as in table type framework or catalog framework. IMO it's slightly better, although I feel there are no short, descriptive, non-overloaded names for that
- only required fields are framework, type, and name: in case where name or type does not make sense, for example because some system only allows to provide you a singular catalog, or where there is only single type of catalog for given framework, it's okay to provide some default value. 
- distinguished between `uri` and `warehouseUri` - the first provides URI of a catalog - for example JDBC connection string - and the second provides the physical location of the data in this catalog. In some cases this does not make sense, like for `delta`, and those fields are optional.
- make the `source` field optional. Yes, it duplicates `processing_engine` facet for Spark, but I think there are cases (like `DatasetEvent` cases) where this would be useful.

mobuchowski commented on website/static/spec/facets/1-0-0/CatalogDatasetFacet.json:
```diff
@@ -0,0 +1,56 @@
+              "type": "string",
+              "example": "my_iceberg_catalog"
+            },
+            "uri": {
```
Ok, done.

dolfinus commented:
LGTM, thanks @mobuchowski
